File: data/transcoder_evaluation_gfg/c/UGLY_NUMBERS.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

unsigned f_gold ( unsigned n ) {
  unsigned ugly [ n ];
  unsigned i2 = 0, i3 = 0, i5 = 0;
  unsigned next_multiple_of_2 = 2;
  unsigned next_multiple_of_3 = 3;
  unsigned next_multiple_of_5 = 5;
  unsigned next_ugly_no = 1;
  ugly [ 0 ] = 1;
  for ( int i = 1;
  i < n;
  i ++ ) {
    next_ugly_no = min ( next_multiple_of_2, min ( next_multiple_of_3, next_multiple_of_5 ) );
    ugly [ i ] = next_ugly_no;
    if ( next_ugly_no == next_multiple_of_2 ) {
      i2 = i2 + 1;
      next_multiple_of_2 = ugly [ i2 ] * 2;
    }
    if ( next_ugly_no == next_multiple_of_3 ) {
      i3 = i3 + 1;
      next_multiple_of_3 = ugly [ i3 ] * 3;
    }
    if ( next_ugly_no == next_multiple_of_5 ) {
      i5 = i5 + 1;
      next_multiple_of_5 = ugly [ i5 ] * 5;
    }
  }
  return next_ugly_no;
}


unsigned f_filled ( unsigned n ) {}

int main(void) {
    int n_success = 0;
    int param0[] = {27,64,93,90,85,86,72,86,32,1};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i]) == f_gold(param0[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/rust/UGLY_NUMBERS.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut n: u32) -> u32 {
    let vla = n as usize;
    let mut ugly: Vec<u32> = ::std::vec::from_elem(0, vla);
    let mut i2: u32 = 0 as i32 as u32;
    let mut i3: u32 = 0 as i32 as u32;
    let mut i5: u32 = 0 as i32 as u32;
    let mut next_multiple_of_2: u32 =
        2 as i32 as u32;
    let mut next_multiple_of_3: u32 =
        3 as i32 as u32;
    let mut next_multiple_of_5: u32 =
        5 as i32 as u32;
    let mut next_ugly_no: u32 = 1 as i32 as u32;
    *ugly.as_mut_ptr().offset(0 as i32 as isize) =
        1 as i32 as u32;
    let mut i: i32 = 1 as i32;
    while (i as u32) < n {
        next_ugly_no =
            min(next_multiple_of_2 as i32,
                min(next_multiple_of_3 as i32,
                    next_multiple_of_5 as i32)) as u32;
        *ugly.as_mut_ptr().offset(i as isize) = next_ugly_no;
        if next_ugly_no == next_multiple_of_2 {
            i2 = i2.wrapping_add(1 as i32 as u32);
            next_multiple_of_2 =
                (*ugly.as_mut_ptr().offset(i2 as
                                               isize)).wrapping_mul(2 as
                                                                        i32
                                                                        as
                                                                        u32)
        }
        if next_ugly_no == next_multiple_of_3 {
            i3 = i3.wrapping_add(1 as i32 as u32);
            next_multiple_of_3 =
                (*ugly.as_mut_ptr().offset(i3 as
                                               isize)).wrapping_mul(3 as
                                                                        i32
                                                                        as
                                                                        u32)
        }
        if next_ugly_no == next_multiple_of_5 {
            i5 = i5.wrapping_add(1 as i32 as u32);
            next_multiple_of_5 =
                (*ugly.as_mut_ptr().offset(i5 as
                                               isize)).wrapping_mul(5 as
                                                                        i32
                                                                        as
                                                                        u32)
        }
        i += 1
    }
    return next_ugly_no;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [27 as i32, 64 as i32, 93 as i32,
         90 as i32, 85 as i32, 86 as i32,
         72 as i32, 86 as i32, 32 as i32,
         1 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize] as u32) ==
               f_gold(param0[i as usize] as u32) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
